auth/postgres/init.go

```diff
@@ -54,39 +54,31 @@ func migrateDB(db *sqlx.DB) error {
 						expires_at  TIMESTAMP,
 						PRIMARY KEY (id, issuer_id)
 					)`,
-					`CREATE extension LTREE`,
-					`CREATE TABLE IF NOT EXISTS group_type (
-						id INTEGER UNIQUE NOT NULL,
-						name VARCHAR(254) UNIQUE NOT NULL,
-						PRIMARY KEY (id)
-					)`,
+					`CREATE EXTENSION IF NOT EXISTS LTREE`,
 					`CREATE TABLE IF NOT EXISTS groups ( 
 						id          VARCHAR(254) UNIQUE NOT NULL,
 						parent_id   VARCHAR(254), 
 						owner_id    VARCHAR(254),
 						name        VARCHAR(254) NOT NULL,
 						description VARCHAR(1024),
 						metadata    JSONB,
-						path        LTREE, 
-						type        INTEGER NOT NULL,
+						path        LTREE,
 						created_at  TIMESTAMPTZ,
 						updated_at  TIMESTAMPTZ,
-						PRIMARY KEY (owner_id, path),
-						FOREIGN KEY (parent_id) REFERENCES groups (id) ON DELETE CASCADE,
-						FOREIGN KEY (type) REFERENCES group_type (id)
+						UNIQUE (owner_id, name, parent_id),
```

Comment:
Do we need `name` here?

Reply:
we are forbiding same name on the given hierarchy level